examples/lang2/dns/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(proc_macro_hygiene)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::storage;
+use ink_lang2 as ink;
+
+#[ink::contract(version = "0.1.0")]
+mod dns {
+    #[ink(storage)]
+    struct DomainNameService {
+        /// A hashmap to store all name to addresses mapping.
+        name_to_address: storage::HashMap<Hash, AccountId>,
+        /// A hashmap to store all name to owners mapping.
+        name_to_owner: storage::HashMap<Hash, AccountId>,
+        /// The default address.
+        default_address: storage::Value<AccountId>,
+    }
+
+    /// Emitted whenever a new name is being registered.
+    #[ink(event)]
+    struct Register {
+        #[ink(topic)]
+        name: Hash,
+        #[ink(topic)]
+        from: AccountId,
+    }
+
+    /// Emitted whenever an address changes.
+    #[ink(event)]
+    struct SetAddress {
+        #[ink(topic)]
+        name: Hash,
+        from: AccountId,
+        #[ink(topic)]
+        old_address: Option<AccountId>,
+        #[ink(topic)]
+        new_address: AccountId,
+    }
+
+    /// Emitted whenver a name is being transferred.
+    #[ink(event)]
+    struct Transfer {
+        #[ink(topic)]
+        name: Hash,
+        from: AccountId,
+        #[ink(topic)]
+        old_owner: Option<AccountId>,
+        #[ink(topic)]
+        new_owner: AccountId,
+    }
+
+    impl DomainNameService {
+        /// Creates a new domain name service contract.
+        #[ink(constructor)]
+        fn new(&mut self) {
+            self.default_address.set(AccountId::from([0x00; 32]));
```

Comment:
Dumb question, but couldn't we use `Default::default()`? Seems like it would simplify the code and be more readable as well.